tests/test_clients.py

```diff
@@ -112,11 +114,44 @@ async def test_title_search(paper_attributes: dict[str, str]) -> None:
             ),
         )
         details = await client.query(title=paper_attributes["title"])
+
+        # matches the citation pattern, not the specific citation count
+        # search because the count will be somewhere in the middle of the string
+        expected_citation_format = re.search(
+            DocDetails.CITATION_COUNT_REGEX_PATTERN,
+            paper_attributes["formatted_citation"],
+        )
+        actual_citation_format = re.search(
+            DocDetails.CITATION_COUNT_REGEX_PATTERN,
+            details.formatted_citation,  # type: ignore[union-attr]
+        )
+        assert (
+            expected_citation_format is not None
+        ), "Expected string should match the citation pattern"
+        assert (
+            actual_citation_format is not None
+        ), "Actual string should match the citation pattern"
```

Comment:
```suggestion
        assert (
            expected_citation_format
        ), "Expected string should match the citation pattern"
        assert (
            actual_citation_format
        ), "Actual string should match the citation pattern"
```

This may resolve the `union-attr` ignore a few lines below